### DISCUSSION TITLE
Bad temperature Thermostat

### DISCUSSION THREAD

**Discussion by Hood93** [milo526/homebridge-tuya-web#169] [Q&A] [answered]
Hi,

this plugin is fantastic! Thank you.
Yesterday I added my thermostat!
 to the home app. It works well, I can turn on ad off the thermostat. But I see bad internal temperature, like this screenshot. How can I resolve it? 
 
 [7BD563C2-E6EF-47FA-AD33-2715D410E881](https://user-images.githubusercontent.com/64242888/104415472-5e338700-5572-11eb-9704-fde9da6743fe.png)

**Comment by milo526** [comment]
Please see the section on configuring devices in the ReadMe.

https://github.com/milo526/homebridge-tuya-web#thermostatclimate

**Reply by Hood93** [reply]
Hi,

I read this:

"Thermostat/Climate

These devices can device a minimum- and maximum temperature as well as a temperature factor.

The minimum and maximum values must be entered as degrees celsius with half degree increments; i.e. -16, 5, 23, 32.5.
This will influence the minimum and maximum temperature that you will be able to set the thermostat at in HomeKit.

The temperature factor can be used to influence the shown temperature. If HomeKit is showing an extremely high temperature, please try setting this value to 0.1. This will change the shown value from i.e. 220 to 220 * 0.1 = 22. The value entered here must be a positive decimal value i.e. 0.1, 1, 2.5.

If desired, you can overwrite these devices to the temperature sensor. This will only report the current temperature and not allow you to change the temperature."

But in my home there are 15.5 degree, as indicate from tuya app, but in home app I see 33

![image](https://user-images.githubusercontent.com/64242888/104422200-b8394a00-557c-11eb-8d98-b895d5ce69fc.png)

Thank you

**Comment by milo526** [comment, marked as answer]
So then try `0.5` since `33 * 0.5 = 16.5` and that seems fairly logical.

You probably have a temperature offset configured in which you tell tuya that the temperature is actual 1 degree cooler then the measured temperature. This is currently not possible through this plugin.

So try setting both the temperature factors to 0.5. easiest is too do this through the visual configuration from config-ui-x

**Reply by Hood93** [reply]
Thank you! 
I try this at home! 
I'll give you feedback!

**Reply by Hood93** [reply]
Thank you Milo, i modify just now the value, and I see correct temperature!
